nosnoc/problem.py

```diff
@@ -599,6 +624,70 @@ def add_fe_constraints(self, fe: FiniteElement, ctrl_idx: int):
         self.ind_comp[ctrl_idx].append(increment_indices(fe.ind_comp, g_len))
         return
 
+    def create_complementarity(self, x: List[ca.SX], y: ca.SX, sigma: ca.SX, tau: ca.SX) -> None:
```

Comment:
I don't like how this is duplicated from `FiniteElement`.

What about just gathering the complementarity pairs at FE level and then only using the function here?

I will actually also need the list of the complementarity pairs, in my current custom solver WIP implementation.